docs/reference/sql/functions.md

```diff
@@ -11,5 +15,140 @@ functions in DataFusion. These functions include:
 In summary, GreptimeDB supports all SQL aggregate functions and scalar functions in DataFusion. Users can safely
 use these rich built-in functions in GreptimeDB to manipulate and analyze data.
 
-# GreptimeDB Functions
-TODO @dennis
+### `arrow_cast`
+
+`arrow_cast` function is from DataFusion's [`arrow_cast`](https://arrow.apache.org/datafusion/user-guide/sql/scalar_functions.html#arrow-cast). It's illustrated as:
+
+```sql
+arrow_cast(expression, datatype)
+```
+
+Where the `datatype` can be any valid Arrow data type in this [list](https://arrow.apache.org/datafusion/user-guide/sql/data_types.html). The four timestamp types are:
+- Timestamp(Second, None)
+- Timestamp(Millisecond, None)
+- Timestamp(Microsecond, None)
+- Timestamp(Nanosecond, None)
+
+(Notice that the `None` means the timestamp is timezone naive)
+
+## GreptimeDB Functions
+
+Please refer to [API documentation](https://greptimedb.rs/script/python/rspython/builtins/greptime_builtin/index.html#functions)
+
+## Time and Date
+
+### `date_trunc`
+
+`date_trunc` function follows the same API with PostgreSQL's [`date_trunc`](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC). It's illustrated as:
+
+```sql
+date_trunc(field, source [, time_zone ])
```

Comment:
rename `field` to a more meaningful name?

Reply:
In the PostgreSQL's doc:

> `field` selects to which precision to truncate the input value.

maybe we could change `field` to `precision`